### DISCUSSION TITLE
Need help about DirectionalLight

### DISCUSSION THREAD

**Discussion by Kris-B** [visgl/deck.gl#5817] [Q&A]
Hi,
Can someone explain me how the ```direction``` is coded, for a directional light source ?

```
const directionalLight= new DirectionalLight({
  color: [128, 128, 0],
  intensity: 2.0,
  direction: [0, -100, -100]
});
```

I understand that an array specifying a 3D vector is requested. But I cannot found any explanation about how 3 vector is coded.
Thanks.

**Comment by sclebal** [comment]
direction vector is normalized in the DirectionalLight constructor. As it's normalized, [0, -100, -100] is the same as [0, -1, -1], both normalized vector is [0, -0.707, -0.707].

**Reply by Pessimistress** [reply]
And in geospatial views, x,y,z axes point to east, north and up.
